### PR TITLE
Adds the Chosen Harvester

### PR DESCRIPTION
**Pull request by Robustin**
Post-game is pretty shitty for cultists, even if you think people SHOULD have to sac themselves to Nar-Sie, getting prodded to death by a dozen vengeful harvesters is a shitty way to go. 

Upon summoning Nar-Sie, all (human) cultists will be gibbed and transformed into Chosen Harvesters.

Chosen Harvesters lose disabling smoke but get forcewall and double HP.

They have a very heavy melee attack but it won't work against human(type) mobs. Instead each melee attack against a human will dismember a single random limb. Attacking a target with no limbs will paralyze them. Chosen Harvesters will retain their cult status and their blood tracker will direct them to Nar-Sie so they can bring in their poor limbless victims for sacrifice.

Because the dismemberment attack does only 15 damage, most targets will be fully conscious for the experience. 

Pics in a bit.

### CODE REVIEW

**Comment by Armhulen** [comment]
cool, but playing as a harvester is pretty shitty too, do you think you can fix that?

**Comment by Robustin** [comment]
As soon as I come up with a way to track which harvester accomplishes a sacrifice, I plan to let a harvester become "Chosen" after 2 sacrifices.

**Comment by Niknakflak** [comment]
I mean yea, but this doesnt seem any more fun, if anything, it seems even less fun than what currently happens

**Comment by Armhulen** [comment]
Also rename it so I can triumphantly update the wiki with a new construct type

**Comment by ExcessiveUseOfCobblestone** [comment]
I enjoy trying to woo the harvesters to not sacrifice me despite being in decked out narsie gear.

**Comment by Robustin** [comment]
@niknakflak

If you don't think it's more fun hacking off someone's limbs one by one and dragging them back screaming to your God isn't more fun than being a useless farting shit I can't help you.

This is mostly for cultists, harvesters can't be the greatest thing ever or people wouldn't even bother fleeing Narsie. Cultists can now stay in cult chat and talk about the round and actually feel like a predator.

I do have vague plans to make the sacrifices more meaningfulike but my priority is to nother make victory feel so shitty for cultists.

File: code/game/gamemodes/cult/runes.dm [hunk 1]
```diff
@@ -492,12 +492,15 @@ structure_check() searches for nearby cultist structures required for the invoca
...
+	cult_mode.eldergod = 0
 	sleep(40)
 	if(src)
 		color = "#FF0000"
-	if(cult_mode)
-		cult_mode.eldergod = 0
-	new /obj/singularity/narsie/large(T) //Causes Nar-Sie to spawn even if the rune has been removed
+	GLOB.blood_target = new /obj/singularity/narsie/large(T) //Causes Nar-Sie to spawn even if the rune has been removed
+	for(var/datum/mind/cult_mind in cult)
+		if(istype(cult_mind.current, /mob/living/carbon/human))
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: ishuman(cult_mind.current)

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 2]
```diff
@@ -286,6 +286,51 @@
...
+	environment_smash = 3
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	construct_spells = list(/obj/effect/proc_holder/spell/aoe_turf/conjure/wall,
+							/obj/effect/proc_holder/spell/aoe_turf/conjure/floor,
+							/obj/effect/proc_holder/spell/aoe_turf/conjure/lesserforcewall)
+	playstyle_string = "<B>You are a Chosen Harvester. You have razor-sharp pincers, the ability to repair your bretheren, and an arsenal of magic: \
+						Find those who still cling to this illusory, cut them to pieces, and bring them back to the Geometer so they may know Truth.</B>"
+
+/mob/living/simple_animal/hostile/construct/harvester/chosen/AttackingTarget()
+	if(istype(target, /mob/living/carbon/human))
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/constructs.dm]: ishuman(target)

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 3]
```diff
@@ -286,6 +286,51 @@
...
+		var/mob/living/carbon/human/H = target
+		var/list/parts = list()
+		parts = H.bodyparts.Copy(3,0)
+		var/obj/item/bodypart/RIP
+		do_attack_animation(H)
+		if(LAZYLEN(parts) == 0)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/constructs.dm]: if(!LAZYLEN(parts))

**Comment by KorPhaeron** [comment]
Can we just make the round end within 60 seconds of Nar-Sie showing up

Or just make everyone random constructs if they click on nar-sie so its fun without violating LORE

**Comment by KorPhaeron** [comment]
>getting prodded to death by a dozen vengeful harvesters is a shitty way to go.

That's the point! You did something bad and now you are suffering for it

**Comment by KorPhaeron** [comment]
I admit I feel pretty dumb arguing so hard about lore that I'm not sure other people care about at all though

@tgstation/commit-access 

Has my brain stopped working if nobody else cares this much about the theme I'll just drop it

**Comment by ChangelingRain** [comment]
Make all the harvesters have forcewall+this dismemberment effect instead of making it cultist only

**Comment by Robustin** [comment]
You still get gibbed and forcibly constructed Kor, it's not a luxury cruise. Id rather keep my kick ass body if I had the choice, but this is my compromise between "Narsie doesn't love me" and "this is an objectively terrible experience".

**Comment by KorPhaeron** [comment]
The best way to make the experience less objectively terrible would be to make it last 60 seconds tops and then have a cinematic like malf/nuke but getting anyone to animate even something simple will be a nightmare

File: code/game/gamemodes/cult/runes.dm [hunk 4]
```diff
@@ -492,12 +492,15 @@ structure_check() searches for nearby cultist structures required for the invoca
...
+	cult_mode.eldergod = 0
 	sleep(40)
 	if(src)
 		color = "#FF0000"
-	if(cult_mode)
-		cult_mode.eldergod = 0
-	new /obj/singularity/narsie/large(T) //Causes Nar-Sie to spawn even if the rune has been removed
+	GLOB.blood_target = new /obj/singularity/narsie/large(T) //Causes Nar-Sie to spawn even if the rune has been removed
+	for(var/datum/mind/cult_mind in SSticker.mode.cult)
+		if(istype(cult_mind.current, /mob/living/carbon/human))
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/runes.dm]: ishuman(cult_mind.current)

**Comment by AnturK** [comment]
I agree with Kor, Nar-Sie not giving a fuck about cultist status is thematic.

**Comment by coiax** [comment]
image

**Comment by coiax** [comment]
When Nar'sie is summoned, the story of the round isn't about the cultists anymore, it's about the surviving people attempting to flee from the eldritch horror consuming the station.

**Comment by coiax** [comment]
Although, we really need to make a way for harvesters to be able to see through their fart smoke.

**Comment by Niknakflak** [comment]
No I mean, being a human cultist after nar-sie is summoned is more fun than being instantly gibbed and forced to be a harvester anyway

**Comment by Thunder12345** [comment]
Yeah, even the cultists should get to experience the attempt to flee from a wave of harvesters beating them to death and dragging them back to narsie to be eaten

**Comment by factoryman942** [comment]
Does the limbs thing exclude the head?

**Comment by InsaneHyena** [comment]
I don't care about the lore, I just want a fun endgame for cultists.

**Comment by Supermichael777** [comment]
Change their in game name to Reapers. ALL GLORY TO THE BLOOD HARVIST

**Comment by CPTANT** [comment]
This pr is kinda meh but harvester are pretty badly designed. 

Harvesters are completely worthless against anyone wearing internals, which makes escaping the station extremely boring.

**Comment by KorPhaeron** [comment]
@CPTANT 

They're both unfun to play because they can't hurt anything and unfun to fight because they just dismantle the shuttle I agree its trash for everyone involved.

7 fucking minutes of it as well

**Comment by Thunder12345** [comment]
They can hurt plenty, you just need to act like a swarm of locusts and not a lone wolf.

**Comment by ChangelingRain** [comment]
Really, we should remove harvester wallsmash and give them incorporeal move and an aoe turf conversion ability instead.

**Comment by Iamgoofball** [comment]
# make cult summons end the game like the nuke 2k17

**Comment by Robustin** [comment]
1) This is still an objective improvement over what we have now.

2) Nothing about this violates even Kor's belief in the lore, Kor's just throwing up a strawman. **Nar-Sie creating "chosen" harvesters doesn't mean Nar-Sie's now a benevolent and merciful god who wants to help its followers.** We've established that Nar-Sie likes sacrifices, hence the existence of harvesters - and we've established that harvesters are controlled by what remains of the souls who have been sacrificed. From there it's perfectly acceptable to argue that Nar-Sie would empower those who have proven their competence through service and (successfully) summoning it with a more powerful form to seek out and destroy more hardened targets (aka anyone wearing a mask). There isn't compassion in Nar-Sie's action, it it a cold logic designed to bring in more sacrifices. 

3) Once again I've got a good PR that, when its not fighting strawmen, falls into the category of "the greatest enemy of a good plan is the dream of a perfect plan". If someone wants to draw up a round-end animation for the cult(s) then go nuts - we can make all harvesters strong and enjoy a minute or two of slaughter before the animation plays. 

Alternatively I'd like to hear actual arguments against this compared to our current system and not the fictional system in your mind that you will never actually help code. 

Like I said I have further plans for the post-summoning period but this was an obvious and necessary first step because no post-summon gameplay is going to be decent if its just people in masks preying on harvesters preying on people without masks.

**Comment by dannno** [comment]
I don't see the point when narsie means the round is over anyway
this seems like a pointless change, I'd rather have the lore retained than add this

**Comment by Robustin** [comment]
Except the rounds not over, there's literally 6 more minutes after summoning.

Also I still don't hear an argument for this being "against lore". I specifically dumped my original idea for this one because of "lore" memes. I made it as inoffensive to lore as possible by making sure all the cultists still die, everyone gets turned into harvesters, harvesters still focus on bringing crew to Nar-Sie, etc.

**Pull request closed without merging**